SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp

```diff
@@ -114,15 +117,19 @@ void FRPCContainer::ProcessRPCs(FArrayOfParams& RPCList)
 	int NumProcessedParams = 0;
 	for (auto& Params : RPCList)
 	{
-		if (ApplyFunction(Params))
+		ERPCQueueCommand QueueCommand = ApplyFunction(Params);
+		switch (QueueCommand)
 		{
+		case ERPCQueueCommand::ContinueProcessing:
 			NumProcessedParams++;
-		}
-		else
-		{
+		case ERPCQueueCommand::DropEntireQueue:
+			RPCList.Empty();
+			return;
+		case ERPCQueueCommand::DropRPC:
```

Comment:
This doesn't look right. Previously, when `bShouldDrop` was true, `ApplyFunction` would return `true` so this would keep processing the RPCs. We really have these cases:
* RPC executed successfully, keep processing
* RPC failed, keep processing (drop RPC)
* RPC failed, queue it
* RPC failed, drop entire queue

Reply:
In fact, for dropping RPC on the sending side, it looks like we would actually end up dropping the entire queue in most cases:
* target object was destroyed
* we don't have an actor channel
* (for legacy RPCs) we don't have authority and `WillHaveAuthorityOverActor` returns false
The only remaining case is `MissingFunctionInfo`, and that one looks quite odd to me. I can't imagine a situation where you call an RPC but the function is missing. I guess we can just drop it and keep processing the remaining RPCs in the queue just in case.